actionview/lib/action_view/template/renderable.rb

```diff
@@ -4,20 +4,22 @@ module ActionView
   class Template
     # = Action View Renderable Template for objects that respond to #render_in
     class Renderable # :nodoc:
-      def initialize(renderable)
+      def initialize(renderable, formats, &block)
         @renderable = renderable
+        @formats = formats
+        @block = block
       end
 
       def identifier
         @renderable.class.name
       end
 
-      def render(context, *args)
-        @renderable.render_in(context)
+      def render(context, locals)
+        @renderable.render_in(context, formats: @formats, locals: locals, &@block)
```

Comment:
This is a breaking change for those objets. We probably need to deprecate the old signature and tell people to upgrade their objects

Reply:
@rafaelfranca I could also see Rails checking the arity?

Regardless, it'd be a huge bummer to have a breaking change like this. I'd _really_ hope we could avoid going down that route.

Reply:
Yeah, I was thinking on doing the arity check to show the deprecation.

Why would it be a bummer if we deprecated the old signature? The old code would work, showing a deprecation, and if the library changed to the new one, as the arguments are optional, things would just work in old versions of Rails. In fact you can release a new version of a library right now with the new signature.

Reply:
If this is a breaking change, would it need to be part of a major version like 7.2 to 8.0?

If not, is the deprecation in the current implementation sufficient enough for a future minor release (like 8.0 to 8.1)?